### ISSUE TITLE
QuickSpec doesn't prune with respect to eta reduction

### ISSUE THREAD

**Issue by isovector** [nick8325/quickspec#40]
I'm getting the following result:

```
== Functions ==                      
subst1 :: Subst1 -> LexemeVW' () () -> [LexemeVW' () ()]

== Laws ==
  1. subst1 s >=> subst1 s = subst1 s
  2. (subst1 s >=> subst1 s) lvw = subst1 s lvw
```

(2) is exactly (1), except eta expanded!

**Comment by nick8325** [comment]
I can't reproduce this. I tried the following example:

```
import QuickSpec
import Control.Monad

something :: Int -> [Int]
something 1 = [2, 2]
something x = [x]

main =
  quickSpec [
    background $ con ">=>" ((>=>) :: (A -> [B]) -> (B -> [C]) -> (A -> [C])),
    con "something" something ]
```

And got:

```
== Functions ==
(>=>) :: (a -> [b]) -> (b -> [c]) -> a -> [c]

== Functions ==                  
something :: Int -> [Int]

== Laws ==
  1. something >=> something = something
```

So there must be something different in your example which is tickling a bug. Do you have a test case you can share?

**Comment by isovector** [comment]
Here's a repro:

https://gist.github.com/isovector/2f66e5b548e67c33a579fa2fc22b6e84

or available in repro form if you'd prefer:

https://github.com/isovector/transliterator/tree/a6a30990476b2e9a0210c57f49d27fa9bc6541f1

The `Spec` module when run produces the follownig output:

```text
== Functions ==                                                                                                                              
(>=>) :: (a -> [b]) -> (b -> [c]) -> a -> [c]                                                                                                
                                                                                                                                             
== Functions ==                                                                                                                              
subst1 :: Subst1 -> LexemeVW' () () -> [LexemeVW' () ()]                                                                                     
                                                                                                                                             
WARNING: The following types have no 'Arbitrary' instance declared.                                                                          
You will not get any variables of the following types:                                                                                       
  Subst1 -> [Int]                                                                                                                            
  Subst1 -> [LexemeVW' () ()]                                                                                                                
  LexemeVW' () () -> [Int]                                                                                                                   
  [LexemeVW' () ()] -> [Int]                                                                                                                 
  LexemeVW' () () -> [LexemeVW' () ()]                                                                                                       
  [LexemeVW' () ()] -> [LexemeVW' () ()]                                                                                                     
                                                                                                                                             
== Laws ==                                                                                                                                   
  1. subst1 s >=> subst1 s = subst1 s                                                                                                        
  2. (subst1 s >=> subst1 s2) lvw = (subst1 s2 >=> subst1 s) lvw                                                                             
  3. (subst1 s >=> subst1 s) lvw = subst1 s lvw                                                                                              
  4. (subst1 s >=> subst1 s2) >=> subst1 s = subst1 s >=> subst1 s2  
```

**Comment by nick8325** [comment]
Looks like this was fixed in c6eb8bd!
